Src/AutoFixtureUnitTest/MailAddressGeneratorTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Net.Mail;
+using System.Text.RegularExpressions;
+using Ploeh.AutoFixture;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.AutoFixtureUnitTest.Kernel;
+using Xunit;
+
+namespace Ploeh.AutoFixtureUnitTest
+{
+    public class MailAddressGeneratorTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilder()
+        {
+            // Fixture setup
+            // Exercise system
+            var sut = new MailAddressGenerator();
+            // Verify outcome
+            Assert.IsAssignableFrom<ISpecimenBuilder>(sut);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNullRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var sut = new MailAddressGenerator();
+            var dummyContext = new DelegatingSpecimenContext();
+            // Exercise system
+            var result = sut.Create(null, dummyContext);
+            // Verify outcome
+            Assert.Equal(new NoSpecimen(), result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNonMailAddressRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var request = new object();
+            var sut = new MailAddressGenerator();
+            var dummyContext = new DelegatingSpecimenContext();
+            // Exercise system
+            var result = sut.Create(request, dummyContext);
+            // Verify outcome
+            var expectedResult = new NoSpecimen(request);
+            Assert.Equal(expectedResult, result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNullContextDoesNotThrow()
+        {
+            // Fixture setup
+            var sut = new MailAddressGenerator();
+            var dummyRequest = new object();
+            // Exercise system and verify outcome
+            Assert.DoesNotThrow(() => sut.Create(dummyRequest, null));

```

Comment:
Let's say that we take `MailAddressGenerator` as is (as I'm very close to thinking that we should), but later want to enhance it so that it'll request names and domain names from the context.

If that's the case, the `context` becomes a required parameter. If we keep this test around, it might be interpreted as a breaking change. What if we didn't have this test?